reV/SAM/generation.py

```diff
@@ -34,6 +39,56 @@
 class Generation(RevPySam, ABC):
```

Comment:
Do we want to rename this BaseGeneration since its an abstract class?

Reply:
agree, will rename AbstractSamGeneration